Allow users to list all of their tasks

TomomiKimura opened the pull request:
Resolves #1

![Screen Shot 2022-01-05 at 15 42 13](https://user-images.githubusercontent.com/90229918/148172192-5b2565f6-50f0-4e88-a7bc-25ef8333600f.png)

TomomiKimura committed:
Allow users to list all of their tasks

Resolves #1

MiguelAraCo commented on src/handlers/list.ts:
```diff
@@ -1,10 +1,28 @@
     all: boolean;
   };
 
+  const taskData = "data/db.json";
```
This is probably going to be used by other handlers, we may want to move it to somewhere else

TomomiKimura replied:
I agree! Do you mind if I tackle this on the next PR?

MiguelAraCo replied:
No problem! Let's move it once it's needed!

MiguelAraCo commented on src/handlers/list.ts:
```diff
@@ -1,10 +1,28 @@
+      }
+    });
+    for (const eachTask of taskStrings) {
+      eachTask.length;
```
What is this line doing? Maybe a left over?

TomomiKimura replied:
That's true! Thank you for noticing. I will fix it.

MiguelAraCo commented on src/handlers/list.ts:
```diff
@@ -1,10 +1,28 @@
   export const handle = async (args: Args): Promise<void> => {
-    // TODO
-    throw new Error("Not yet implemented");
+    const data = await readFile(taskData);
+    const stringContent = data.toString();
+    const taskObjects: Array<Task> = JSON.parse(stringContent);
```
What happens if the file is damaged? We may need to handle errors here

TomomiKimura replied:
You are right! I fixed and pushed the changes so that please check!

MiguelAraCo commented on src/handlers/list.ts:
```diff
@@ -1,10 +1,28 @@
+    const taskObjects: Array<Task> = JSON.parse(stringContent);
+
+    const taskStrings: Array<string> = taskObjects.map((task) => {
+      if (task.done) {
+        return `[x] ${task.description}`;
+      } else {
+        return `[ ] ${task.description}`;
+      }
```
We could simplify this to something like:

```suggestion
      return `[${task.done ? 'x' : ' '}] ${task.description}`;
```

TomomiKimura replied:
That's amazing! But I would like to keep the old version so that I can understand what is going on here.. I will keep improving so that I will be able to understand codes like what you suggested! Thank you!

MiguelAraCo reviewed (changes requested):
Great job on your first pull request! Overall looks good but found a couple of things, please take a look at them when you get time.

TomomiKimura committed:
Fixed some parts about PR:
-delete left over(line23)
-add error handling in case the file is damaged(line13-20)

MiguelAraCo reviewed (approved):
LGTM! 🎉
